tests/Data/ScillaTestUtil.cpp

```diff
@@ -0,0 +1,130 @@
+ * code are governed by the GNU General Public License v3.0 (available at
+ * https://www.gnu.org/licenses/gpl-3.0.en.html) ('GPLv3'). The programs that
+ * are governed by GPLv3.0 are those programs that are located in the folders
+ * src/depends and tests/depends and which include a reference to GPLv3 in their
+ * program files.
+ */
+
+#include <boost/filesystem.hpp>
+#include "common/Constants.h"
+
+#include "ScillaTestUtil.h"
+
+using namespace boost::multiprecision;
+
+bool ScillaTestUtil::ParseJsonFile(Json::Value &j, std::string filename) {
+  if (!boost::filesystem::is_regular_file(filename)) return false;
+
+  std::ifstream in(filename, std::ios::binary);
+  std::string fstr;
+
+  fstr = {std::istreambuf_iterator<char>(in), std::istreambuf_iterator<char>()};
+
+  Json::CharReaderBuilder builder;
+  std::unique_ptr<Json::CharReader> reader(builder.newCharReader());
+  std::string errors;
+
+  return reader->parse(fstr.c_str(), fstr.c_str() + fstr.size(), &j, &errors);
+}
+
+// Get ScillaTest for contract "name" and test numbered "i".
+bool ScillaTestUtil::GetScillaTest(ScillaTest &t, std::string contrName,
+                                   unsigned int i) {
+  if (SCILLA_ROOT.empty()) {
+    return false;
+  }
+
+  // TODO: Does this require a separate entry in constants.xml?
+  std::string testDir = SCILLA_ROOT + "/tests/contracts/" + contrName;
+  if (!boost::filesystem::is_directory(testDir)) return false;
+
+  if (!boost::filesystem::is_regular_file(testDir + "/contract.scilla"))
+    return false;
+
+  std::ifstream in(testDir + "/contract.scilla", std::ios::binary);
+  t.code = {std::istreambuf_iterator<char>(in),
+            std::istreambuf_iterator<char>()};
+
+  std::string is(std::to_string(i));
+
+  // Get all JSONs
+  return !(
+      !ParseJsonFile(t.init, testDir + "/init.json") ||
+      !ParseJsonFile(t.state, testDir + "/state_" + is + ".json") ||
+      !ParseJsonFile(t.blockchain, testDir + "/blockchain_" + is + ".json") ||
+      !ParseJsonFile(t.expOutput, testDir + "/output_" + is + ".json") ||
+      !ParseJsonFile(t.message, testDir + "/message_" + is + ".json"));
+}
+
+// Get _balance from output state of interpreter, from OUTPUT_JSON.
+// Return 0 on failure.
+uint128_t ScillaTestUtil::GetBalanceFromOutput(void) {
+  Json::Value iOutput;
+  if (!ScillaTestUtil::ParseJsonFile(iOutput, OUTPUT_JSON)) {
+    LOG_GENERAL(WARNING, "Unable to parse output of interpreter.");
+    return 0;
+  }
+
+  // Get balance as given by the interpreter.
+  uint128_t oBal = 0;
+  Json::Value states = iOutput["states"];
+  for (auto &state : states) {
+    if (state["vname"] == "_balance") oBal = atoi(state["value"].asCString());
+  }
+
+  return oBal;
+}
+
+// Return BLOCKNUMBER in Json. Return 0 if not found.
+uint64_t ScillaTestUtil::GetBlockNumberFromJson(Json::Value &blockchain) {
+  // Get blocknumber from blockchain.json
+  uint64_t bnum = 0;
+  for (auto &it : blockchain)
```

Comment:
sorry bro, it's not enforced but some generic coding fashion among our code base so that making the code easier to read and maintain. Actually myself can help do the change, but put it in review here is just for u to keep informed also.

Reply:
>making the code easier to read and maintain

If this was true, I would happily do it. I can't see how not having curly brackets around single statement for/if statement can make the code less readable or difficult to maintain. I'm actually looking for a healthy discussion around this.

I'm all for making code uniform, readable and maintainable. I just want to make sure that practices aimed at this actually are serving the purpose.

Reply:
yeah, if we have complex for `for` and `if` conditions wrapped leaving out the bracket will surely increase the difficulty in understanding the logic.

Reply:
Sure, but this was a single statement, taking 3 lines totally including the `for`, `if` and the statement inside the `if`.

Reply:
Sure, i know your concern, as I said it's just a general coding fashion, isn't it nice to follow if doesn't cause too many trouble?